include/cuco/static_map.cuh

```diff
@@ -434,14 +434,26 @@ class static_map {
 
     __host__ __device__ device_view_base(pair_atomic_type* slots,
                                          std::size_t capacity,
-                                         Key empty_key_sentinel,
-                                         Value empty_value_sentinel,
-                                         Key erased_key_sentinel) noexcept
+                                         sentinel::empty_key<Key> empty_key_sentinel,
+                                         sentinel::empty_value<Value> empty_value_sentinel) noexcept
       : slots_{slots},
         capacity_{capacity},
-        empty_key_sentinel_{empty_key_sentinel},
-        erased_key_sentinel_{erased_key_sentinel},
-        empty_value_sentinel_{empty_value_sentinel}
+        empty_key_sentinel_{empty_key_sentinel.value},
+        erased_key_sentinel_{empty_key_sentinel.value},
```

Comment:
Ok, so we set the `erased_key_sentinel` to be the same as the `empty_key_sentinel`. This implies, that we must not call `map.erase(...)` or we break our probing scheme. In order to ensure this, we have a guard in the host API [here](https://github.com/PointKernel/cuCollections/blob/76dff03985ad224e068a0077eabeb6cbb26068a9/include/cuco/detail/static_map.inl#L161).
However, as far as I can see, there is no such guard for the device API.
Is this by design?

Reply:
We don't really have a good way to guard it on the device API. The best would could probably do is an `assert`.

Reply:
Agreed. We should at least add a disclaimer to the doxygen docs, which are btw missing for the device-sided `erase(…)` functions [here](https://github.com/NVIDIA/cuCollections/blob/76dff03985ad224e068a0077eabeb6cbb26068a9/include/cuco/static_map.cuh#L934) and [here](https://github.com/NVIDIA/cuCollections/blob/76dff03985ad224e068a0077eabeb6cbb26068a9/include/cuco/static_map.cuh#L940).

Reply:
Docs added.